weather_api/openapi/openapi.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import os
+from pathlib import Path
+from string import Template
+
+from fastapi import FastAPI
+from fastapi.openapi.utils import get_openapi
+from fastapi.routing import APIRoute
+
+from weather_api.config import settings
+
+supported_languages = {"Curl": "curl", "JavaScript": "js", "Python": "py"}
+
+
+def custom_openapi(app: FastAPI, example_code_dir: Path):
+    if app.openapi_schema:
+        return app.openapi_schema
+
+    openapi_schema = get_openapi(
+        title="Weather API",
+        version=settings.version,
+        description=settings.api_description,
+        routes=app.routes,
+    )
+
+    openapi_schema["info"]["x-logo"] = {
+        "url": f"{settings.api_root_path}/assets/open-epi-logo.svg"
+    }
+
+    routes_that_need_doc = [
+        route for route in app.routes if isinstance(route, APIRoute)
+    ]
+    for route in routes_that_need_doc:
+        code_samples = []
+        for lang in supported_languages:
+            file_with_code_sample = (
+                example_code_dir
+                / lang.lower()
+                / f"{route.name}.{supported_languages[lang]}"
+            )
+            if os.path.isfile(file_with_code_sample):
+                with open(file_with_code_sample) as f:
+                    code_template = Template(f.read())
+                    code_samples.append(
+                        {
+                            "lang": lang,
+                            "source": code_template.safe_substitute(
+                                api_host=settings.public_url,
+                                api_root_path=settings.api_root_path,
+                                api_endpoint=route.path,
```

Comment:
Maybe the files can be made a bit simpler if you concatenate the URL here, and only pass in a single variable?